### PR TITLE
AngularFireAuth API's

### PR DESCRIPTION
**Pull request by mukesh51**
Added AngulaFireAuth API's as per issue #93. Please review and let me know, if any changes are needed.

### CODE REVIEW

**Comment by katowulf** [comment]
@mukesh51 for some reason there's no conflict shown here, but [this change](https://github.com/angular/angularfire2/pull/564) should be causing one. I'm making appropriate edits here to match that change so it's not lost.

File: docs/api-reference.md [hunk 1]
```diff
@@ -189,6 +194,64 @@ export class MyApp {
...
+ signInWithFacebook(): Promise<FirebaseAuthState> {
+ 	     return this.auth$.login({
+ 		 provider: AuthProviders.Facebook,
+ 		 method: AuthMethods.Popup,
+ 	     });
+     }
+ ....
+ ```
+ 
+ This is similar to how firebase provides authentication
```
Comment (katowulf) [inline docs/api-reference.md]: Please revise these to just say:  See <link> for more.

File: docs/api-reference.md [hunk 2]
```diff
@@ -226,6 +289,48 @@ class App {
...
+		this.af.auth.logout();
+	}
+```
+
+`createUser(credentials: EmailPasswordCredentials): firebase.Promise<FirebaseAuthState>` : Creates a new user with email/password provided. Returns a promise filled with FirebaseAuthState, which contains the uid of the created user.
+
+The credentials object is an object containing email and password of the user to be created. Similar to [createUserWithEmailAndPassword](https://firebase.google.com/docs/reference/js/firebase.auth.Auth#createUserWithEmailAndPassword)
+
+```ts
+  createNewUser(): Promise<FirebaseAuthState> {
```
Comment (katowulf) [inline docs/api-reference.md]: I think this should be:  `createNewUser( properties:Object ): Promise<FirebaseAuthState> {`?

File: docs/api-reference.md [hunk 3]
```diff
@@ -226,6 +289,48 @@ class App {
...
+            this.authState = state;          
+        });        
+    }
+
+    get authenticated(): boolean {        
+        return this.authState !== null;
+    }
+```
+
+`logout(): void`: Deletes the authentication token issued by Firebase and signs user out. See [Auth.signOut()](https://firebase.google.com/docs/reference/js/firebase.auth.Auth#signOut) in the Firebase API reference.
```
Comment (katowulf) [inline docs/api-reference.md]: Probably worth noting that logout() is an asynchronous operation. There is an open bug against the Firebase SDK to make this return a promise.

File: docs/api-reference.md [hunk 4]
```diff
@@ -226,6 +289,48 @@ class App {
 
 ```
 
+
+`getAuth(): FirebaseAuthState` : Returns the client's current Authentication State.
```
Comment (katowulf) [inline docs/api-reference.md]: Where is FirebaseAuthState documented? Can we link to it?

**Comment by katowulf** [comment]
Hi @mukesh51! Great work! I added a few comments. Please address.

File: docs/api-reference.md (hunk 1 above)
Comment (mukesh51) [inline docs/api-reference.md]: @katowulf Updated as advised.

File: docs/api-reference.md (hunk 2 above)
Comment (mukesh51) [inline docs/api-reference.md]: @katowulf Updated as advised. yes it's a 
createNewUser( properties:Object ): Promise<FirebaseAuthState>

File: docs/api-reference.md (hunk 3 above)
Comment (mukesh51) [inline docs/api-reference.md]: @katowulf Thanks for update. I was unaware that this is an open bug against firebase. I pasted the contents as per your suggestion. Hope that's ok.

File: docs/api-reference.md (hunk 4 above)
Comment (mukesh51) [inline docs/api-reference.md]: FirebaseAuthState is an interface in the definition file auth_backend.d.ts. I added the method signature in documentation, which might be helpful, incase someone is digging for more information. 
I think it is similar to https://firebase.google.com/docs/reference/js/firebase.auth.AuthCredential , but I am not sure, hence didn't link it. Please verify and let me know as well. Thanks.,

**Comment by mukesh51** [comment]
Hi @katowulf . Did you get a chance to review this. Anything missing ??

**Comment by katowulf** [comment]
Great work! Sorry it took a while to get these reviewed.

**Pull request merged**